tests/std/tests/P0896R4_ranges_alg_merge/test.cpp

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <cassert>
+#include <concepts>
+#include <ranges>
+#include <utility>
+
+#include <range_algorithm_support.hpp>
+
+using namespace std;
+using P = pair<int, int>;
+
+// Validate that merge_result aliases in_in_out_result
+STATIC_ASSERT(same_as<ranges::merge_result<int, void*, double>, ranges::in_in_out_result<int, void*, double>>);
+
+// Validate dangling story
+STATIC_ASSERT(same_as<decltype(ranges::merge(borrowed<false>{}, borrowed<false>{}, nullptr_to<int>)),
+    ranges::merge_result<ranges::dangling, ranges::dangling, int*>>);
+STATIC_ASSERT(same_as<decltype(ranges::merge(borrowed<false>{}, borrowed<true>{}, nullptr_to<int>)),
+    ranges::merge_result<ranges::dangling, int*, int*>>);
+STATIC_ASSERT(same_as<decltype(ranges::merge(borrowed<true>{}, borrowed<false>{}, nullptr_to<int>)),
+    ranges::merge_result<int*, ranges::dangling, int*>>);
+STATIC_ASSERT(same_as<decltype(ranges::merge(borrowed<true>{}, borrowed<true>{}, nullptr_to<int>)),
+    ranges::merge_result<int*, int*, int*>>);
+
+struct instantiator {
+    static constexpr P elements1[] = {{0, 10}, {0, 11}, {0, 12}, {1, 10}, {1, 11}, {3, 10}};
+    static constexpr P elements2[] = {{13, 0}, {14, 0}, {10, 2}, {11, 3}, {12, 3}};
+    static constexpr P expected[]  = {
+        {0, 10}, {0, 11}, {0, 12}, {13, 0}, {14, 0}, {1, 10}, {1, 11}, {10, 2}, {3, 10}, {11, 3}, {12, 3}};
+
+    template <ranges::input_range R1, ranges::input_range R2, weakly_incrementable O>
+    static constexpr void call() {
+        using ranges::merge, ranges::merge_result, ranges::end, ranges::equal, ranges::iterator_t, ranges::less,
+            ranges::size;
+
+        { // Validate range overload

```

Comment:
We discussed in the [ranges::remove family](https://github.com/microsoft/STL/pull/1005#discussion_r453018420) PR the idea of testing for the number of comparisons when part of the algorithm is abstracted into another function.  We erred on the side of adding additional test coverage in that case, though I don't feel incredibly strongly about it here.  Just something to consider.

Reply:
Counting comparator added!